Core/include/Acts/Seeding/HoughTransformUtils.ipp

```diff
@@ -323,32 +367,38 @@ Acts::HoughTransformUtils::PeakFinders::IslandsAroundMax<
 template <class identifier_t>
 void Acts::HoughTransformUtils::PeakFinders::IslandsAroundMax<identifier_t>::
     extendMaximum(
-        std::vector<std::pair<std::size_t, std::size_t>>& inMaximum,
-        std::vector<std::pair<std::size_t, std::size_t>>& toExplore,
-        YieldType threshold,
-        std::map<std::pair<std::size_t, std::size_t>, YieldType>& yieldMap) {
+        const Acts::HoughTransformUtils::HoughPlane<identifier_t>& houghPlane,
+        std::vector<std::array<std::size_t, 2>>& inMaximum,
+        std::vector<std::size_t>& toExplore, YieldType threshold,
+        std::unordered_map<std::size_t, YieldType>& yieldMap) {
   // in this call, we explore the last element of the toExplore list.
   // Fetch it and pop it from the vector.
-  auto nextCand = toExplore.back();
+
+  const std::size_t candidate = toExplore.back();
+  YieldType& yield{yieldMap[candidate]};
   toExplore.pop_back();
   // check if we are above threshold. Don't add this cell to the island if not
-  if (yieldMap[nextCand] < threshold) {
+  if (yield < threshold) {
     return;
   }
   // This candidate is above threshold and should go on the island!
-
+  auto nextCand = houghPlane.axisBins(candidate);
   // add it to the cell list for the island
   inMaximum.push_back(nextCand);
   // and "veto" the hit for further use via the yield map
-  yieldMap[nextCand] = -1.0f;
+  yieldMap[candidate] = -1.0f;
 
   // now we have to collect the non empty neighbours of this cell and check them
   // as well
   for (auto step : m_stepDirections) {
-    auto newCand = std::make_pair(nextCand.first + step.first,
-                                  nextCand.second + step.second);
-    // no need for bounds-checking, as the map structure will default to "empty"
-    // yields for OOB
+    std::array newCandxy = {nextCand[0] + step.first,
```

Comment:
unspecificied type and size array, does this compile? 😕

Reply:
wow it does! 🤯

Reply:
the magic of c++17 !